numpy/core/tests/test_numeric.py

```diff
@@ -1149,6 +1149,13 @@ def test_sufficient_width(self):
         assert_equal(np.binary_repr(10, width=7), '0001010')
         assert_equal(np.binary_repr(-5, width=7), '1111011')
 
+    def test_neg_width_boundaries(self):
+        # see gh-8670
+        for width in range(1, 11):
+            num = -2**(width - 1)
+            exp = '1' + (width - 1) * '0'
+            assert_equal(np.binary_repr(num, width=width), exp)
```

Comment:
Basically just add the exact test case from the issue, `numpy.binary_repr(-128, 8)`, so it's obvious that the test is doing the right thing